torch/csrc/jit/codegen/cuda/lower_scalar_hoist.cpp

```diff
@@ -308,7 +337,19 @@ std::pair<int64_t, bool> findAllocPointFromDataDependency(
     Val* value) {
   int64_t pos = -1;
   for (auto i : c10::irange(exprs.size())) {
-    for (auto o : exprs[i]->outputs()) {
+    auto expr = exprs[i];
+    if (auto alloc = dynamic_cast<kir::Allocate*>(expr)) {
+      // Currently this branch is only to handle shared memory address. For
+      // shared memory address, we generate code like `toSmem(T7)`, this does
+      // not need the shared memory tensor to be computed, but it do require T7
+      // to be allocated.
+      auto buffer = alloc->buffer();
+      if (buffer == value ||
+          findRefAsSubexprOf(value, buffer, true) != nullptr) {
```

Comment:
Can you give an example of this `findRefAsSubexrOp` returning a non nullptr?

Reply:
For example, if I have index `i1 = toSmem(T1) + i2`, and `buffer` is `T1`, we should find that `T1` as subexpr of `i1`, and `findRefAsSubexprOf` for this case will return non-nullptr